### PR TITLE
Remove support for Cassandra 3.x and add Cassandra 5.x

### LINKED ISSUE #5160: [Deprecation] Drop support for Cassandra 3.x
3.x is declared EOL https://cassandra.apache.org/_/blog/Apache-Cassandra-3.0.x-and-3.11.x-End-of-Life-Announcement.html

According to https://cassandra.apache.org/_/download.html, support for it will end once 5.x is released, which was aimed ad Dec'23, but as of 2023-12-05 only 5-Beta was released. In any case, I expect by summer 2024 the 3.x support will end, and we should remove it from Jaeger as well.

This doesn't remove any implementation, only the CI testing can drop testing with 3.x (and we should add testing with 5.x. once it is released).

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
-  Part of #5160 

## Description of the changes
- Deprecated Cassandra 3.x as it was marked EOL
- Enabled CI testing and schema generation for 5.x which GA-ed on 09/05 (https://cassandra.apache.org/_/download.html)

## How was this change tested?
- CI

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/storage/cassandra/schema/create.sh [hunk 1]
```diff
@@ -35,6 +35,9 @@ if [[ "$template" == "" ]]; then
         4)
             template=$(dirname $0)/v004.cql.tmpl
             ;;
+        5)
+            template=$(dirname $0)/v005.cql.tmpl
+            ;;
```
Comment (mahadzaryab1) [inline plugin/storage/cassandra/schema/create.sh:40]: Do we want to drop schema generation for v3?

Comment (yurishkuro) [reply plugin/storage/cassandra/schema/create.sh:40]: These are not Cassandra versions but our db schema evolutions. Does something need to change for Cassandra v5?

**Review by yurishkuro — COMMENTED** [review]
So they finally released it. Almost a year late.

File: plugin/storage/cassandra/schema/create.sh (hunk 1 above)
Comment (mahadzaryab1) [reply plugin/storage/cassandra/schema/create.sh:40]: no i just copied the same file to maintain the separation - let me know if you want me to combine it

Comment (yurishkuro) [reply plugin/storage/cassandra/schema/create.sh:40]: If the db schema is not changing there's no need to create a new file.

Comment (mahadzaryab1) [reply plugin/storage/cassandra/schema/create.sh:40]: should i just use the v4 file then?

Comment (yurishkuro) [reply plugin/storage/cassandra/schema/create.sh:40]: Yes - schema version and Cassandra version are not coupled.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
I removed `Resolves` from description because we also need to update documentation as part of that ticket.

**Pull request merged**